nuxeo-java-client-test/src/test/java/org/nuxeo/client/ITRepository.java

```diff
@@ -380,6 +383,40 @@ public void itCanManagePermissions() {
         assertEquals("local", acp.getAcls().get(0).getName());
     }
 
+    @Test
+    public void itCanManagePermissionsOnCustomACL() {
+        // Create a user
+        User user = nuxeoClient.userManager().createUser(ITBase.createUser());
+        // Add a first permission on a new ACL "testPerm"
+        Document folder = nuxeoClient.repository().fetchDocumentByPath("/folder_2");
+        // Settings
+        ACE ace = new ACE();
+        ace.setUsername(user.getUserName());
+        ace.setPermission("Write");
+        ace.setCreator("Administrator");
+        ace.setBlockInheritance(false);
+        folder.addPermission(ace, "testPerm");
+        // Check created permission
+        folder = nuxeoClient.repository().fetchDocumentByPath("/folder_2");
+        ACP acp = folder.fetchPermissions();
+        assertTrue(acp.getAcls().size() != 0);
+        assertEquals(2, acp.getAcls().size());
+        Optional<ACL> testPermACL = acp.getAcls()
+                                       .stream()
+                                       .filter(a -> StringUtils.equals("testPerm", a.getName()))
+                                       .findFirst();
+        assertTrue(testPermACL.isPresent());
```

Comment:
If you prefer (so not a requested change), you can use the orElseThrow method on the Optional, like below:
```suggestion
        ACL testPermACL = acp.getAcls()
                             .stream()
                             .filter(a -> StringUtils.equals("testPerm", a.getName()))
                             .findFirst();
                             .orElseThrow(() -> new AssertionError("ACL with name: testPerm should exist"))
```
As it you can get rid of Optional and later call on get.

Reply:
Nice! I will do the change